src/main/java/sky/command/DeleteCommand.java

```diff
@@ -0,0 +1,56 @@
+package sky.command;
+
+import java.io.IOException;
+
+import sky.TaskList;
+import sky.exception.TextNoMeaningException;
+import sky.storage.History;
+import sky.storage.Storage;
+import sky.task.Task;
+
+/**
+ * The DeleteCommand class deals with deleting a task from taskList.
+ */
+public class DeleteCommand extends Command {
+    private String fullCommand;
+
+    public DeleteCommand(String fullCommand) {
+        this.fullCommand = fullCommand;
+    }
+
+    @Override
+    public String execute(TaskList taskList, Storage storage, History history)
+            throws TextNoMeaningException, IOException {
+        try {
+            int taskNum = generateTaskNumToDelete();
+            Task task = taskList.getTask(taskNum);
+            taskList.removeTask(task);
+            storage.reWriteDataFile(taskList);
+            history.addHistoryInTime(taskList);
+            return generateResponse(task, taskList);
```

Comment:
I like the way you organised the code into different levels of abstraction for better readability! 👍